Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```diff
@@ -102,103 +94,79 @@ public CONNECTSamlAssertionValidator(PropertyAccessor propertyAccessor) {
-            validators.add(org.opensaml.Configuration.getValidatorSuite("saml2-core-schema-validator"));
-            validators.addAll(getSaml2SpecValidators());
-
-            for (AuthzDecisionStatement auth : assertion.getSaml2().getAuthzDecisionStatements()) {
-                if (StringUtils.isBlank(auth.getResource())) {
-                    auth.setResource(TEMP_RESOURCE_FOR_VALIDATION);
+            final Issuer issuer = assertion.getSaml2().getIssuer();
+            validateIssuer(issuer);
+            try {
+                // need to process custom saml validator
+                for (final Saml2ExchangeAuthFrameworkValidator validator : getSaml2SpecValidators()) {
+                    validator.validateAssertion(assertion);
                 }
+            } catch (final WSSecurityException e) {
+                LOG.error("Saml Validation error: " + e.getMessage(), e);
+                throw new WSSecurityException(ErrorCode.FAILED_CHECK, "invalidSAMLsecurity");
             }
+        }
+    }
 
-            Issuer issuer = assertion.getSaml2().getIssuer();
-            if (issuer.getFormat().equals("urn:oasis:names:tc:SAML:1.1:nameid-format:entity")) {
-                if (!StringUtils.isBlank(issuer.getSPProvidedID())) {
-                    throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
-                            + "urn:oasis:names:tc:SAML:1.1:nameid-format:entity" + "" + "and"
-                            + "Security/Assertion/Issuer/@SPProvidedID" + " " + "is present.");
-                }
-                if (!StringUtils.isBlank(issuer.getNameQualifier())) {
-                    throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
-                            + "urn:oasis:names:tc:SAML:1.1:nameid-format:entity" + "" + "and"
-                            + "Security/Assertion/Issuer/@NameQualifier" + " " + "is present.");
-                }
-
-                if (!StringUtils.isBlank(issuer.getSPNameQualifier())) {
-                    throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
-                            + "urn:oasis:names:tc:SAML:1.1:nameid-format:entity" + "" + "and"
-                            + "Security/Assertion/Issuer/@SPNameQualifier" + " " + "is present.");
-
-                }
+    /**
+     * @param issuer
+     * @throws WSSecurityException
+     */
+    private static void validateIssuer(Issuer issuer) throws WSSecurityException {
+        if (SamlConstants.URN_OASIS_NAMES.equals(issuer.getFormat())) {
+            if (StringUtils.isNotBlank(issuer.getSPProvidedID())) {
+                throw new WSSecurityException(ErrorCode.FAILED_CHECK,
+                    SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT + " " + SamlConstants.URN_OASIS_NAMES + " " + "and"
+                        + SamlConstants.SECURITY_ASSERTION_ISSUER + " " + IS_PRESENT);
```

Comment:
don't use string+string to concatenate.  Use stringBuilder instead.  Verify if we can reduce this duplication string